scripts/dependency.sh

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -eou pipefail
+set -x
+
+export LC_ALL=C
+
+PWD=$(cd "$(dirname "$0")"/.. || exit; pwd)
+
+function printUsage() {
+  echo "Usage: $(basename "${0}") [-p <artifactId>] -r " 2>&1
+  echo '   -r   [OPTIONAL] to replace the old dependencyList file with new dependencies'
+  echo '   -p   [MUST] to generate new dependencyList file for the specified module'
+}
```

Comment:
is the flow for PR owners to run the script manually and submit the diff along with the PR? it could be hard to execute though

Reply:
Yes, the practice that comes from the Kyuubi community is like this.

If the owner of PR changed the dependencies of maven. The contributor must install the project in its local. And run this script locally to override these dependency list files and submit the changes.

For Github workflow, we only add a new one to run this script and compare with the PR, if they are different, then show the diff and fail the workflow.